src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java

```diff
@@ -9,9 +9,10 @@
 import org.myrobotlab.framework.MRLListener;
 import org.myrobotlab.framework.MethodEntry;
 import org.myrobotlab.framework.Outbox;
+import org.myrobotlab.service.interfaces.ServiceLifeCycleListener;
 
 public interface ServiceInterface
-    extends ServiceQueue, LoggingSink, NameTypeProvider, MessageSubscriber, MessageSender, StateSaver, Invoker, StatePublisher, StatusPublisher, ServiceStatus, Attachable {
+    extends ServiceLifeCycleListener, ServiceQueue, LoggingSink, NameTypeProvider, MessageSubscriber, MessageSender, StateSaver, Invoker, StatePublisher, StatusPublisher, ServiceStatus, Attachable {

```

Comment:
seems like the service interface should implement the ServiceLifeCycle interface... unless you're saying generically that a service can generically listen to another services lifecycle..  is that what you're intending here?

Reply:
correct - the abstract base class Service now implements a set of callbacks,
the "interested" services can simply override these if they want to handle the notification events of
registered, created, started, stopped and released.
If they are not interested, they simply do not override.